components/ws-manager-api/typescript/src/client-provider-source.ts

```diff
@@ -23,12 +23,19 @@ export interface WorkspaceManagerClientProviderSource {
 export class WorkspaceManagerClientProviderEnvSource implements WorkspaceManagerClientProviderSource {
     protected _clusters: WorkspaceCluster[] | undefined = undefined;
 
-    public async getWorkspaceCluster(name: string, applicationCluster: string): Promise<WorkspaceCluster | undefined> {
-        return this.clusters.find((m) => m.name === name && m.applicationCluster === applicationCluster);
+    public async getWorkspaceCluster(name: string): Promise<WorkspaceCluster | undefined> {
+        return this.clusters.find((m) => m.name === name);
```

Comment:
Why use a different filter here than in the next method, `getAllWorkspaceClusters`? IMO makes it impossible to understand without a comment.

Reply:
Also, when making region part of the PrimaryKey, we should adjust the interface `WorkspaceManagerClientProviderSource`.